diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7252975..b751077 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,9 +11,12 @@ macro(build_yaml_cpp)
   if(DEFINED CMAKE_BUILD_TYPE)
     list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
   endif()
+  if(DEFINED CMAKE_TOOLCHAIN_FILE)
+    list(APPEND extra_cmake_args -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE})
+  endif()
   if(NOT WIN32)
-    list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=-std=c++14 -w")
-    list(APPEND extra_cmake_args "-DCMAKE_C_FLAGS=-w")
+	  #list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=-std=c++14 -w")
+	  #list(APPEND extra_cmake_args "-DCMAKE_C_FLAGS=-w")
   endif()
 
   list(APPEND extra_cmake_args "-DYAML_CPP_BUILD_TESTS=OFF")
