diff --git a/rosidl_typesupport_c/CMakeLists.txt b/rosidl_typesupport_c/CMakeLists.txt
index 765a7ad..2732bde 100644
--- a/rosidl_typesupport_c/CMakeLists.txt
+++ b/rosidl_typesupport_c/CMakeLists.txt
@@ -17,8 +17,8 @@ endif()

 find_package(ament_cmake_ros REQUIRED)
 # provides FindPoco.cmake and Poco on platforms without it
-find_package(poco_vendor)
-find_package(Poco COMPONENTS Foundation)
+#find_package(poco_vendor)
+#find_package(Poco COMPONENTS Foundation)
 find_package(rosidl_generator_c REQUIRED)

 link_directories(${Poco_LIBRARY_DIR})
diff --git a/rosidl_typesupport_c/cmake/rosidl_typesupport_c_generate_interfaces.cmake b/rosidl_typesupport_c/cmake/rosidl_typesupport_c_generate_interfaces.cmake
index 61ccafe..fef4517 100644
--- a/rosidl_typesupport_c/cmake/rosidl_typesupport_c_generate_interfaces.cmake
+++ b/rosidl_typesupport_c/cmake/rosidl_typesupport_c_generate_interfaces.cmake
@@ -142,8 +142,8 @@ else()
       "linking was requested")
   endif()
   if(NOT rosidl_typesupport_c_SUPPORTS_POCO)
-    message(FATAL_ERROR "Multiple typesupports [${typesupports}] but Poco was "
-      "not available when rosidl_typesupport_c was built")
+	  #message(FATAL_ERROR "Multiple typesupports [${typesupports}] but Poco was "
+	  #"not available when rosidl_typesupport_c was built")
   endif()
 endif()
 
