diff --git a/rclcpp/minimal_action_client/CMakeLists.txt b/rclcpp/minimal_action_client/CMakeLists.txt
index 1dab0e6..4057ffc 100644
--- a/rclcpp/minimal_action_client/CMakeLists.txt
+++ b/rclcpp/minimal_action_client/CMakeLists.txt
@@ -14,24 +14,29 @@ find_package(ament_cmake REQUIRED)
 find_package(example_interfaces REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rclcpp_action REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 add_executable(action_client_not_composable not_composable.cpp)
 ament_target_dependencies(action_client_not_composable
   "rclcpp"
   "rclcpp_action"
-  "example_interfaces")
+  "example_interfaces"
+  "yaml")
 
 add_executable(action_client_not_composable_with_cancel not_composable_with_cancel.cpp)
 ament_target_dependencies(action_client_not_composable_with_cancel
   "rclcpp"
   "rclcpp_action"
-  "example_interfaces")
+  "example_interfaces"
+  "yaml")
 
 add_executable(action_client_not_composable_with_feedback not_composable_with_feedback.cpp)
 ament_target_dependencies(action_client_not_composable_with_feedback
   "rclcpp"
   "rclcpp_action"
-  "example_interfaces")
+  "example_interfaces"
+  "yaml")
 
 install(TARGETS
   action_client_not_composable
diff --git a/rclcpp/minimal_action_server/CMakeLists.txt b/rclcpp/minimal_action_server/CMakeLists.txt
index a4c488e..a20b3c2 100644
--- a/rclcpp/minimal_action_server/CMakeLists.txt
+++ b/rclcpp/minimal_action_server/CMakeLists.txt
@@ -14,12 +14,15 @@ find_package(ament_cmake REQUIRED)
 find_package(example_interfaces REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rclcpp_action REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 add_executable(action_server_not_composable not_composable.cpp)
 ament_target_dependencies(action_server_not_composable
   "rclcpp"
   "rclcpp_action"
-  "example_interfaces")
+  "example_interfaces"
+  "yaml")
 
 install(TARGETS action_server_not_composable
   DESTINATION lib/${PROJECT_NAME})
diff --git a/rclcpp/minimal_client/CMakeLists.txt b/rclcpp/minimal_client/CMakeLists.txt
index eb51bef..0671174 100644
--- a/rclcpp/minimal_client/CMakeLists.txt
+++ b/rclcpp/minimal_client/CMakeLists.txt
@@ -13,8 +13,12 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(example_interfaces REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
+link_directories(${CMAKE_PREFIX_PATH}/lib)
 add_executable(client_main main.cpp)
+target_link_libraries(client_main yaml)
 ament_target_dependencies(client_main rclcpp example_interfaces)
 
 install(TARGETS client_main
diff --git a/rclcpp/minimal_composition/CMakeLists.txt b/rclcpp/minimal_composition/CMakeLists.txt
index f856e43..7bf0140 100644
--- a/rclcpp/minimal_composition/CMakeLists.txt
+++ b/rclcpp/minimal_composition/CMakeLists.txt
@@ -14,6 +14,8 @@ find_package(ament_cmake REQUIRED)
 find_package(class_loader REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(std_msgs REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 include_directories(include)
 
@@ -32,18 +34,19 @@ if(NOT WIN32)
     )
 endif()
 
+link_directories(${CMAKE_PREFIX_PATH}/lib)
 add_executable(composition_publisher src/standalone_publisher.cpp)
-target_link_libraries(composition_publisher composition_nodes)
+target_link_libraries(composition_publisher composition_nodes yaml)
 ament_target_dependencies(composition_publisher
   rclcpp composition_nodes)
 
 add_executable(composition_subscriber src/standalone_subscriber.cpp)
-target_link_libraries(composition_subscriber composition_nodes)
+target_link_libraries(composition_subscriber composition_nodes yaml)
 ament_target_dependencies(composition_subscriber
   rclcpp composition_nodes)
 
 add_executable(composition_composed src/composed.cpp)
-target_link_libraries(composition_composed composition_nodes)
+target_link_libraries(composition_composed composition_nodes yaml)
 ament_target_dependencies(composition_composed rclcpp class_loader)
 
 install(TARGETS
diff --git a/rclcpp/minimal_publisher/CMakeLists.txt b/rclcpp/minimal_publisher/CMakeLists.txt
index 3344f14..35e9ae1 100644
--- a/rclcpp/minimal_publisher/CMakeLists.txt
+++ b/rclcpp/minimal_publisher/CMakeLists.txt
@@ -13,14 +13,20 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(std_msgs REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
+link_directories(${CMAKE_PREFIX_PATH}/lib)
 add_executable(publisher_lambda lambda.cpp)
+target_link_libraries(publisher_lambda yaml)
 ament_target_dependencies(publisher_lambda rclcpp std_msgs)
 
 add_executable(publisher_member_function member_function.cpp)
+target_link_libraries(publisher_member_function yaml)
 ament_target_dependencies(publisher_member_function rclcpp std_msgs)
 
 add_executable(publisher_not_composable not_composable.cpp)
+target_link_libraries(publisher_not_composable yaml)
 ament_target_dependencies(publisher_not_composable rclcpp std_msgs)
 
 install(TARGETS
diff --git a/rclcpp/minimal_service/CMakeLists.txt b/rclcpp/minimal_service/CMakeLists.txt
index d474c49..ca9cea9 100644
--- a/rclcpp/minimal_service/CMakeLists.txt
+++ b/rclcpp/minimal_service/CMakeLists.txt
@@ -13,9 +13,11 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(example_interfaces REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 add_executable(service_main main.cpp)
-ament_target_dependencies(service_main rclcpp example_interfaces)
+ament_target_dependencies(service_main rclcpp example_interfaces yaml)
 
 install(TARGETS service_main
   DESTINATION lib/${PROJECT_NAME})
diff --git a/rclcpp/minimal_subscriber/CMakeLists.txt b/rclcpp/minimal_subscriber/CMakeLists.txt
index bcff7fa..b6ffd12 100644
--- a/rclcpp/minimal_subscriber/CMakeLists.txt
+++ b/rclcpp/minimal_subscriber/CMakeLists.txt
@@ -13,15 +13,17 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(std_msgs REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 add_executable(subscriber_lambda lambda.cpp)
-ament_target_dependencies(subscriber_lambda rclcpp std_msgs)
+ament_target_dependencies(subscriber_lambda rclcpp std_msgs yaml)
 
 add_executable(subscriber_member_function member_function.cpp)
-ament_target_dependencies(subscriber_member_function rclcpp std_msgs)
+ament_target_dependencies(subscriber_member_function rclcpp std_msgs yaml)
 
 add_executable(subscriber_not_composable not_composable.cpp)
-ament_target_dependencies(subscriber_not_composable rclcpp std_msgs)
+ament_target_dependencies(subscriber_not_composable rclcpp std_msgs yaml)
 
 install(TARGETS
   subscriber_lambda
diff --git a/rclcpp/minimal_timer/CMakeLists.txt b/rclcpp/minimal_timer/CMakeLists.txt
index 72be805..9ac46a5 100644
--- a/rclcpp/minimal_timer/CMakeLists.txt
+++ b/rclcpp/minimal_timer/CMakeLists.txt
@@ -12,12 +12,14 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 add_executable(timer_lambda lambda.cpp)
-ament_target_dependencies(timer_lambda rclcpp)
+ament_target_dependencies(timer_lambda rclcpp yaml)
 
 add_executable(timer_member_function member_function.cpp)
-ament_target_dependencies(timer_member_function rclcpp)
+ament_target_dependencies(timer_member_function rclcpp yaml)
 
 install(TARGETS
   timer_lambda
