diff --git a/pendulum_control/CMakeLists.txt b/pendulum_control/CMakeLists.txt
index 8ef16e0..3ab98e3 100644
--- a/pendulum_control/CMakeLists.txt
+++ b/pendulum_control/CMakeLists.txt
@@ -22,6 +22,9 @@ find_package(rclcpp REQUIRED)
 find_package(pendulum_msgs REQUIRED)
 find_package(rttest)
 find_package(tlsf_cpp)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
+
 
 include_directories(include)
 
@@ -36,21 +39,27 @@ ament_target_dependencies(pendulum_demo${target_suffix}
   "pendulum_msgs"
   "rclcpp${target_suffix}"
   "rttest"
-  "tlsf_cpp")
+  "tlsf_cpp"
+  "yaml"
+)
 
 add_executable(pendulum_logger${target_suffix}
   src/pendulum_logger.cpp)
 ament_target_dependencies(pendulum_logger${target_suffix}
   "pendulum_msgs"
   "rclcpp${target_suffix}"
-  "rttest")
+  "rttest"
+  "yaml"
+)
 
 add_executable(pendulum_teleop${target_suffix}
   src/pendulum_teleop.cpp)
 ament_target_dependencies(pendulum_teleop${target_suffix}
   "pendulum_msgs"
   "rclcpp${target_suffix}"
-  "rttest")
+  "rttest"
+  "yaml"
+)
 
 install(TARGETS
   pendulum_demo${target_suffix}
