diff --git a/dynamixel_sdk/include/dynamixel_sdk/port_handler.h b/dynamixel_sdk/include/dynamixel_sdk/port_handler.h
index 63976c0..6d577e2 100644
--- a/dynamixel_sdk/include/dynamixel_sdk/port_handler.h
+++ b/dynamixel_sdk/include/dynamixel_sdk/port_handler.h
@@ -22,7 +22,7 @@
 #ifndef DYNAMIXEL_SDK_INCLUDE_DYNAMIXEL_SDK_PORTHANDLER_H_
 #define DYNAMIXEL_SDK_INCLUDE_DYNAMIXEL_SDK_PORTHANDLER_H_
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #define WINDECLSPEC
 #elif defined(__APPLE__)
 #define WINDECLSPEC
diff --git a/dynamixel_sdk/src/dynamixel_sdk/group_bulk_read.cpp b/dynamixel_sdk/src/dynamixel_sdk/group_bulk_read.cpp
index bcea755..eac87f1 100644
--- a/dynamixel_sdk/src/dynamixel_sdk/group_bulk_read.cpp
+++ b/dynamixel_sdk/src/dynamixel_sdk/group_bulk_read.cpp
@@ -19,7 +19,7 @@
 #include <stdio.h>
 #include <algorithm>
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #include "group_bulk_read.h"
 #elif defined(__APPLE__)
 #include "group_bulk_read.h"
@@ -242,4 +242,4 @@ bool GroupBulkRead::getError(uint8_t id, uint8_t* error)
   {
     return false;
   }
-}
\ No newline at end of file
+}
diff --git a/dynamixel_sdk/src/dynamixel_sdk/group_bulk_write.cpp b/dynamixel_sdk/src/dynamixel_sdk/group_bulk_write.cpp
index 665378f..c1c40c5 100644
--- a/dynamixel_sdk/src/dynamixel_sdk/group_bulk_write.cpp
+++ b/dynamixel_sdk/src/dynamixel_sdk/group_bulk_write.cpp
@@ -18,7 +18,7 @@
 
 #include <algorithm>
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #include "group_bulk_write.h"
 #elif defined(__APPLE__)
 #include "group_bulk_write.h"
diff --git a/dynamixel_sdk/src/dynamixel_sdk/group_sync_read.cpp b/dynamixel_sdk/src/dynamixel_sdk/group_sync_read.cpp
index f392943..b2164df 100644
--- a/dynamixel_sdk/src/dynamixel_sdk/group_sync_read.cpp
+++ b/dynamixel_sdk/src/dynamixel_sdk/group_sync_read.cpp
@@ -18,7 +18,7 @@
 
 #include <algorithm>
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #include "group_sync_read.h"
 #elif defined(__APPLE__)
 #include "group_sync_read.h"
@@ -211,4 +211,4 @@ bool GroupSyncRead::getError(uint8_t id, uint8_t* error)
   {
     return false;
   }
-}
\ No newline at end of file
+}
diff --git a/dynamixel_sdk/src/dynamixel_sdk/group_sync_write.cpp b/dynamixel_sdk/src/dynamixel_sdk/group_sync_write.cpp
index 1202d47..436440b 100644
--- a/dynamixel_sdk/src/dynamixel_sdk/group_sync_write.cpp
+++ b/dynamixel_sdk/src/dynamixel_sdk/group_sync_write.cpp
@@ -18,7 +18,7 @@
 
 #include <algorithm>
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #include "group_sync_write.h"
 #elif defined(__APPLE__)
 #include "group_sync_write.h"
diff --git a/dynamixel_sdk/src/dynamixel_sdk/packet_handler.cpp b/dynamixel_sdk/src/dynamixel_sdk/packet_handler.cpp
index c926715..aa6a0ce 100644
--- a/dynamixel_sdk/src/dynamixel_sdk/packet_handler.cpp
+++ b/dynamixel_sdk/src/dynamixel_sdk/packet_handler.cpp
@@ -16,7 +16,7 @@
 
 /* Author: zerom, Ryu Woon Jung (Leon) */
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #include "packet_handler.h"
 #include "protocol1_packet_handler.h"
 #include "protocol2_packet_handler.h"
diff --git a/dynamixel_sdk/src/dynamixel_sdk/port_handler.cpp b/dynamixel_sdk/src/dynamixel_sdk/port_handler.cpp
index 3054c85..82ffda7 100644
--- a/dynamixel_sdk/src/dynamixel_sdk/port_handler.cpp
+++ b/dynamixel_sdk/src/dynamixel_sdk/port_handler.cpp
@@ -16,7 +16,7 @@
 
 /* Author: zerom, Ryu Woon Jung (Leon) */
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #include "port_handler.h"
 #include "port_handler_linux.h"
 #elif defined(__APPLE__)
diff --git a/dynamixel_sdk/src/dynamixel_sdk/protocol1_packet_handler.cpp b/dynamixel_sdk/src/dynamixel_sdk/protocol1_packet_handler.cpp
index 2525968..bd51d53 100644
--- a/dynamixel_sdk/src/dynamixel_sdk/protocol1_packet_handler.cpp
+++ b/dynamixel_sdk/src/dynamixel_sdk/protocol1_packet_handler.cpp
@@ -16,7 +16,7 @@
 
 /* Author: zerom, Ryu Woon Jung (Leon) */
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #include "protocol1_packet_handler.h"
 #elif defined(__APPLE__)
 #include "protocol1_packet_handler.h"
diff --git a/dynamixel_sdk/src/dynamixel_sdk/protocol2_packet_handler.cpp b/dynamixel_sdk/src/dynamixel_sdk/protocol2_packet_handler.cpp
index c9d90c6..6e80f41 100755
--- a/dynamixel_sdk/src/dynamixel_sdk/protocol2_packet_handler.cpp
+++ b/dynamixel_sdk/src/dynamixel_sdk/protocol2_packet_handler.cpp
@@ -16,7 +16,7 @@
 
 /* Author: zerom, Ryu Woon Jung (Leon) */
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__VXWORKS__)
 #include <unistd.h>
 #include "protocol2_packet_handler.h"
 #elif defined(__APPLE__)
