From d32753644361e318077cc6faaaad6075f40f6ba3 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Wed, 8 Apr 2020 21:57:23 +0200
Subject: [PATCH 2/3] strnlen s

---
 rcl/src/rcl/security_directory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rcl/src/rcl/security_directory.c b/rcl/src/rcl/security_directory.c
index 6e7d9f5..85f9b1f 100644
--- a/rcl/src/rcl/security_directory.c
+++ b/rcl/src/rcl/security_directory.c
@@ -103,7 +103,7 @@ static bool get_best_matching_directory(
       goto cleanup;
     }
     if (file.is_dir) {
-      size_t matched_name_length = strnlen(file.name, sizeof(file.name) - 1);
+      size_t matched_name_length = strnlen_s(file.name, sizeof(file.name) - 1);
       if (0 ==
         strncmp(file.name, node_name,
         matched_name_length) && matched_name_length > max_match_length)
-- 
2.17.1

