From f0d8296fdc6e346b08d7b30951ae744353216faa Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Thu, 9 Apr 2020 19:19:49 +0200
Subject: [PATCH 3/3] add libyaml vendor

---
 rcl_yaml_param_parser/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rcl_yaml_param_parser/CMakeLists.txt b/rcl_yaml_param_parser/CMakeLists.txt
index 95143c8..1be2dff 100644
--- a/rcl_yaml_param_parser/CMakeLists.txt
+++ b/rcl_yaml_param_parser/CMakeLists.txt
@@ -54,6 +54,7 @@ if(BUILD_TESTING)
 endif()
 
 ament_export_dependencies(ament_cmake)
+ament_export_dependencies(libyaml_vendor)
 ament_export_include_directories(include)
 install(
   DIRECTORY include/
-- 
2.17.1

