From 08cbae832ae43f8a8fabb87b544f0f1597b0dcbc Mon Sep 17 00:00:00 2001
From: "Dragusu, Mihai" <Mihai.Dragusu@windriver.com>
Date: Mon, 20 Jan 2020 17:25:34 +0200
Subject: [PATCH 2/2] add read buffer fix

Signed-off-by: Dragusu, Mihai <Mihai.Dragusu@windriver.com>
---
 src/hlds_laser_publisher.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/hlds_laser_publisher.cpp b/src/hlds_laser_publisher.cpp
index 08a9d98..d719039 100644
--- a/src/hlds_laser_publisher.cpp
+++ b/src/hlds_laser_publisher.cpp
@@ -99,7 +99,12 @@ void LFCDLaser::poll(sensor_msgs::msg::LaserScan::SharedPtr scan)
         // Now that entire start sequence has been found, read in the rest of the message
         got_scan = true;
 
-        read(fd,&raw_bytes[2],2518);
+        int buff_i = 2;
+        while (buff_i < 2518)
+        {
+            ret = read(fd,&raw_bytes[buff_i],2518 - buff_i);
+            buff_i += ret;
+        }
         scan->angle_increment = (2.0*M_PI/360.0);
         scan->angle_min = 0.0;
         scan->angle_max = 2.0*M_PI-scan->angle_increment;
@@ -136,7 +141,6 @@ void LFCDLaser::poll(sensor_msgs::msg::LaserScan::SharedPtr scan)
             }
           }
         }
-
         scan->time_increment = motor_speed/good_sets/1e8;
       }
       else
-- 
2.25.0

