diff --git a/turtlebot3_node/CMakeLists.txt b/turtlebot3_node/CMakeLists.txt
index 3f5c36b..cc6311f 100644
--- a/turtlebot3_node/CMakeLists.txt
+++ b/turtlebot3_node/CMakeLists.txt
@@ -28,6 +28,8 @@ find_package(std_srvs REQUIRED)
 find_package(tf2 REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(turtlebot3_msgs REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 ################################################################################
 # Build
@@ -73,7 +75,7 @@ ament_target_dependencies(${PROJECT_NAME}_lib ${DEPENDENCIES})
 set(EXECUTABLE_NAME "turtlebot3_ros")
 
 add_executable(${EXECUTABLE_NAME} src/node_main.cpp)
-target_link_libraries(${EXECUTABLE_NAME} ${PROJECT_NAME}_lib)
+target_link_libraries(${EXECUTABLE_NAME} ${PROJECT_NAME}_lib "yaml")
 ament_target_dependencies(${EXECUTABLE_NAME} ${DEPENDENCIES})
 
 ################################################################################
