From d8c2aaec1c53c1d4c1021b21d6c2a97f070d3bf3 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Fri, 31 Jan 2025 12:12:42 +0100
Subject: [PATCH] add eigen3 include

---
 orocos_kdl_vendor/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/orocos_kdl_vendor/CMakeLists.txt b/orocos_kdl_vendor/CMakeLists.txt
index de526d6..b4908f0 100644
--- a/orocos_kdl_vendor/CMakeLists.txt
+++ b/orocos_kdl_vendor/CMakeLists.txt
@@ -23,6 +23,7 @@ ament_vendor(orocos_kdl_vendor
     -DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}
     -DENABLE_TESTS:BOOL=OFF
     -DENABLE_EXAMPLES:BOOL=OFF
+    -DEIGEN3_INCLUDE_DIR=${CMAKE_INSTALL_PREFIX}/include/eigen3
   PATCHES patches
 )
 
-- 
2.34.1

