From eb143be4f1a0208a403aaddc49f175c1b72e961c Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Fri, 27 Jan 2023 14:27:55 +0100
Subject: [PATCH 1/2] add __VXWORKS__ to find

---
 src/find.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/find.c b/src/find.c
index d5dbf7a..94cdfcd 100644
--- a/src/find.c
+++ b/src/find.c
@@ -60,7 +60,7 @@ rcutils_find_lastn(const char * str, char delimiter, size_t string_length)
     return SIZE_MAX;
   }
 
-#if defined(_GNU_SOURCE)
+#if defined(_GNU_SOURCE) && !defined(__VXWORKS__)
   const char * ptr = memrchr(str, delimiter, string_length);
   if (ptr == NULL) {
     return SIZE_MAX;
-- 
2.34.1

