From 26b156d1de661ad05b409cc45ebc12ec6b80b455 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Fri, 27 Jan 2023 14:28:26 +0100
Subject: [PATCH 2/2] add __VXWORKS__ to strerror

---
 src/strerror.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/strerror.c b/src/strerror.c
index a425010..d1c2c7e 100644
--- a/src/strerror.c
+++ b/src/strerror.c
@@ -27,7 +27,7 @@ rcutils_strerror(char * buffer, size_t buffer_length)
 {
 #if defined(_WIN32)
   strerror_s(buffer, buffer_length, errno);
-#elif defined(_GNU_SOURCE) && (!defined(ANDROID) || __ANDROID_API__ >= 23) && !defined(__QNXNTO__)
+#elif defined(_GNU_SOURCE) && (!defined(ANDROID) || __ANDROID_API__ >= 23) && !defined(__QNXNTO__) && !defined(__VXWORKS__)
   /* GNU-specific */
   char * msg = strerror_r(errno, buffer, buffer_length);
   if (msg != buffer) {
-- 
2.34.1

