From 512a1e2a7b916b82b27f02275bffccd945cad337 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Sat, 3 Jul 2021 10:13:10 +0200
Subject: [PATCH] apply patch

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7a9e709..dd19402 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,12 @@ macro(build_console_bridge)
   if(DEFINED CMAKE_BUILD_TYPE)
     list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
   endif()
+  if(DEFINED CMAKE_MODULE_PATH)
+    list(APPEND extra_cmake_args -DCMAKE_MODULE_PATH=${CMAKE_MODULE_PATH})
+  endif()
+  if(DEFINED BUILD_TESTING)
+    list(APPEND extra_cmake_args -DBUILD_TESTING=${BUILD_TESTING})
+  endif()
   if(DEFINED CMAKE_TOOLCHAIN_FILE)
     list(APPEND extra_cmake_args "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
     if(ANDROID)
-- 
2.25.1

