diff --git a/CMakeLists.txt b/CMakeLists.txt
index 091e1b9..23401a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,8 @@ find_package(urdf REQUIRED)
 find_package(urdfdom_headers REQUIRED)
 find_package(kdl_parser REQUIRED)
 find_package(tf2_ros REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 include_directories(SYSTEM ${Eigen_INCLUDE_DIRS})
 include_directories(
@@ -46,7 +48,8 @@ target_link_libraries(
   ${sensor_msgs_LIBRARIES}
   ${tf2_ros_LIBRARIES}
   ${urdf_LIBRARIES}
-  ${urdfdom_LIBRARIES})
+  ${urdfdom_LIBRARIES}
+  yaml)
 
 add_library(
   joint_state_listener
