From 2f1b0336f3fa34654bd16d32f09255debc90d103 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Fri, 17 Nov 2023 16:25:22 +0100
Subject: [PATCH] add rmw REQUIRED

---
 rmw_implementation/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rmw_implementation/CMakeLists.txt b/rmw_implementation/CMakeLists.txt
index fc0bdfc..393f665 100644
--- a/rmw_implementation/CMakeLists.txt
+++ b/rmw_implementation/CMakeLists.txt
@@ -13,6 +13,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif()
 
 find_package(ament_cmake REQUIRED)
+find_package(rmw REQUIRED)
 find_package(rmw_implementation_cmake REQUIRED)
 
 if(BUILD_TESTING)
-- 
2.34.1

