From f56e4f5edd66d55dc7714efc312488472a3b87ad Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Sat, 11 Feb 2023 15:23:13 +0100
Subject: [PATCH 1/2] add rmw REQUIRED

---
 rmw_implementation/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rmw_implementation/CMakeLists.txt b/rmw_implementation/CMakeLists.txt
index 2926c71..728db0b 100644
--- a/rmw_implementation/CMakeLists.txt
+++ b/rmw_implementation/CMakeLists.txt
@@ -12,6 +12,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif()
 
 find_package(ament_cmake REQUIRED)
+find_package(rmw REQUIRED)
 find_package(rmw_implementation_cmake REQUIRED)
 
 if(BUILD_TESTING)
-- 
2.34.1

