From 5de430b1512a9bcb33271f831513869c9612c3e8 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Sat, 27 Aug 2022 19:54:04 +0200
Subject: [PATCH] add rmw REQUIRED

---
 rmw_implementation/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rmw_implementation/CMakeLists.txt b/rmw_implementation/CMakeLists.txt
index a54ca52..cbc31f5 100644
--- a/rmw_implementation/CMakeLists.txt
+++ b/rmw_implementation/CMakeLists.txt
@@ -13,6 +13,7 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(ament_index_cpp REQUIRED)
+find_package(rmw REQUIRED)
 find_package(rmw_implementation_cmake REQUIRED)
 
 if(BUILD_TESTING)
-- 
2.34.1

