From 551e2e5d5dccac47346b953e428531a337122215 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Mon, 12 Feb 2024 10:39:25 +0100
Subject: [PATCH 01/10] skip find package THREADS

---
 cmake/modules/FindThirdpartyBoost.cmake | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/cmake/modules/FindThirdpartyBoost.cmake b/cmake/modules/FindThirdpartyBoost.cmake
index 163a91ade..750b785e9 100644
--- a/cmake/modules/FindThirdpartyBoost.cmake
+++ b/cmake/modules/FindThirdpartyBoost.cmake
@@ -11,9 +11,11 @@ set(THIRDPARTY_BOOST_INCLUDE_DIR
     "Path to thirdparty/boost"
 )
 
-find_package(Threads REQUIRED)
+if(NOT VXWORKS)
+    find_package(Threads REQUIRED)
+endif()
 
-if(WIN32 OR APPLE OR ANDROID OR QNXNTO)
+if(WIN32 OR APPLE OR ANDROID OR QNXNTO OR VXWORKS)
     set(THIRDPARTY_BOOST_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT})
 else() # Posix
     set(THIRDPARTY_BOOST_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT} rt)
-- 
2.34.1

