From ac6ccd5057b4b3c6b1fde8c294328708c1ab1e20 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Sun, 2 Jul 2023 17:10:38 +0200
Subject: [PATCH 02/10] skip find package THREADS

---
 src/cpp/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index a5213c259..50db6a8a6 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -364,7 +364,7 @@ else()
     set(HAVE_STRICT_REALTIME 0)
 endif()
 
-if(NOT ANDROID)
+if(NOT ANDROID AND NOT VXWORKS)
     find_package(Threads REQUIRED)
 endif()
 
-- 
2.34.1

