From 7ae096b6049892e15d3753b5a09bf002dbca32dd Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Mon, 12 Feb 2024 10:45:07 +0100
Subject: [PATCH 02/10] skip find package THREADS

---
 src/cpp/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index c82588394..7cae6b7d1 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -383,7 +383,7 @@ else()
     set(HAVE_STRICT_REALTIME 0)
 endif()
 
-if(NOT ANDROID)
+if(NOT ANDROID AND NOT VXWORKS)
     find_package(Threads REQUIRED)
 endif()
 
-- 
2.34.1

