From ccbc67215d5b2fda0f05959e9b69ee78089e4452 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Wed, 17 Feb 2021 21:38:35 +0100
Subject: [PATCH] remove CMAKE_DL_LIBS

---
 src/cpp/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index bb4769345..8218c099d 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -426,7 +426,10 @@ elseif(NOT EPROSIMA_INSTALLER)
 
     # Link library to external libraries.
     target_link_libraries(${PROJECT_NAME} ${PRIVACY} fastcdr foonathan_memory
-        ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS}
+        ${CMAKE_THREAD_LIBS_INIT}
+if(NOT VXWORKS)
+        ${CMAKE_DL_LIBS}
+endif()
         ${TINYXML2_LIBRARY}
         $<$<BOOL:${LINK_SSL}>:OpenSSL::SSL$<SEMICOLON>OpenSSL::Crypto>
         $<$<BOOL:${WIN32}>:iphlpapi$<SEMICOLON>Shlwapi>
-- 
2.25.1

