From 7e34e35714152fd16e4553a99b9de8be4c4bcc8e Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Mon, 13 Dec 2021 14:04:56 +0100
Subject: [PATCH 04/10] set munmap to 0

---
 src/cpp/rtps/persistence/sqlite3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpp/rtps/persistence/sqlite3.c b/src/cpp/rtps/persistence/sqlite3.c
index 89faea5b2..f0d7e2e83 100644
--- a/src/cpp/rtps/persistence/sqlite3.c
+++ b/src/cpp/rtps/persistence/sqlite3.c
@@ -34559,7 +34559,7 @@ static struct unix_syscall {
 #define osMmap ((void*(*)(void*,size_t,int,int,int,off_t))aSyscall[22].pCurrent)
 
 #if !defined(SQLITE_OMIT_WAL) || SQLITE_MAX_MMAP_SIZE>0
-  { "munmap",       (sqlite3_syscall_ptr)munmap,          0 },
+  { "munmap",       (sqlite3_syscall_ptr)0,          0 },
 #else
   { "munmap",       (sqlite3_syscall_ptr)0,               0 },
 #endif
-- 
2.34.1

