From 63b24cb0bf2f1208517d84336ee96188fc770933 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Wed, 27 Aug 2025 22:06:43 +0200
Subject: [PATCH 04/10] fix undefined TIOCOUTQ

---
 src/cpp/rtps/transport/TCPChannelResource.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/cpp/rtps/transport/TCPChannelResource.cpp b/src/cpp/rtps/transport/TCPChannelResource.cpp
index 2fbaa3118..d0d7c2c9c 100644
--- a/src/cpp/rtps/transport/TCPChannelResource.cpp
+++ b/src/cpp/rtps/transport/TCPChannelResource.cpp
@@ -16,6 +16,7 @@
 
 #include <chrono>
 #include <thread>
+#include <sys/ttycom.h>
 
 #include <fastrtps/utils/IPLocator.h>
 #include <rtps/transport/asio_helpers.hpp>
-- 
2.34.1

