From 849859157bb14475db0123a163eb50142687fee0 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Mon, 13 Dec 2021 14:07:47 +0100
Subject: [PATCH 06/10] add include mutex

---
 include/fastrtps/utils/TimedMutex.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/fastrtps/utils/TimedMutex.hpp b/include/fastrtps/utils/TimedMutex.hpp
index b1982f56e..647948deb 100644
--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -27,7 +27,7 @@
 extern int clock_gettime(
         int,
         struct timespec* tv);
-#elif _GTHREAD_USE_MUTEX_TIMEDLOCK
+#elif _GTHREAD_USE_MUTEX_TIMEDLOCK || !defined(__linux__)
 #include <mutex>
 #else
 #include <pthread.h>
@@ -183,6 +183,7 @@ private:
     _Mtx_t mutex_;
 };
 #elif _GTHREAD_USE_MUTEX_TIMEDLOCK || !defined(__unix__)
+#include <mutex>
 using TimedMutex = std::timed_mutex;
 using RecursiveTimedMutex = std::recursive_timed_mutex;
 #else
-- 
2.34.1

