From 76dce30d1580e5b336242ce5acf960e34de7abd3 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Fri, 17 Nov 2023 15:31:55 +0100
Subject: [PATCH 06/10] add include mutex

---
 include/fastrtps/utils/TimedMutex.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/fastrtps/utils/TimedMutex.hpp b/include/fastrtps/utils/TimedMutex.hpp
index d20fae9cb..ac304a514 100644
--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -33,7 +33,7 @@ extern int clock_gettime(
         struct timespec* tv);
 #endif // if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
 
-#elif _GTHREAD_USE_MUTEX_TIMEDLOCK
+#elif _GTHREAD_USE_MUTEX_TIMEDLOCK || !defined(__linux__)
 #include <mutex>
 #else
 #include <pthread.h>
@@ -197,6 +197,7 @@ private:
 #endif // if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
 
 #elif _GTHREAD_USE_MUTEX_TIMEDLOCK || !defined(__unix__)
+#include <mutex>
 using TimedMutex = std::timed_mutex;
 using RecursiveTimedMutex = std::recursive_timed_mutex;
 #else
-- 
2.34.1

