From 9923917dc330b29856d490743d6b4150156fde6d Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Mon, 13 Dec 2021 14:08:55 +0100
Subject: [PATCH 08/10] add EOF

---
 thirdparty/nlohmann-json/json.hpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/thirdparty/nlohmann-json/json.hpp b/thirdparty/nlohmann-json/json.hpp
index e821c79a3..0d5ec1e31 100644
--- a/thirdparty/nlohmann-json/json.hpp
+++ b/thirdparty/nlohmann-json/json.hpp
@@ -6204,7 +6204,7 @@ class lexer : public lexer_base<BasicJsonType>
             switch (get())
             {
                 // end of file while parsing string
-                case std::char_traits<char_type>::eof():
+                case EOF:
                 {
                     error_message = "invalid string: missing closing quote";
                     return token_type::parse_error;
@@ -6793,7 +6793,7 @@ class lexer : public lexer_base<BasicJsonType>
                     {
                         case '\n':
                         case '\r':
-                        case std::char_traits<char_type>::eof():
+                        case EOF:
                         case '\0':
                             return true;
 
@@ -6810,7 +6810,7 @@ class lexer : public lexer_base<BasicJsonType>
                 {
                     switch (get())
                     {
-                        case std::char_traits<char_type>::eof():
+                        case EOF:
                         case '\0':
                         {
                             error_message = "invalid comment; missing closing '*/'";
@@ -7523,7 +7523,7 @@ scan_number_done:
             // end of input (the null byte is needed when parsing from
             // string literals)
             case '\0':
-            case std::char_traits<char_type>::eof():
+            case EOF:
                 return token_type::end_of_input;
 
             // error
-- 
2.34.1

