From 726c849a5c2e5a0c000407137d18a13a07228848 Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@gmail.com>
Date: Wed, 23 Sep 2020 12:57:22 +0200
Subject: [PATCH] added support for Dinkum C library

---
 rclcpp/include/rclcpp/function_traits.hpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/rclcpp/include/rclcpp/function_traits.hpp b/rclcpp/include/rclcpp/function_traits.hpp
index f5c56b0..2ef7d64 100644
--- a/rclcpp/include/rclcpp/function_traits.hpp
+++ b/rclcpp/include/rclcpp/function_traits.hpp
@@ -89,6 +89,8 @@ struct function_traits<std::_Bind<std::_Mem_fn<ReturnTypeT (ClassT::*)(Args ...)
 #elif defined _MSC_VER  // MS Visual Studio
 struct function_traits<
   std::_Binder<std::_Unforced, ReturnTypeT (ClassT::*)(Args ...), FArgs ...>>
+#elif defined _CPPLIB_VER  // Dinkumware C++ library
+struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT(__cdecl ClassT::*)(Args ...), FArgs ...>>
 #else
 #error "Unsupported C++ compiler / standard library"
 #endif
@@ -106,6 +108,8 @@ struct function_traits<std::_Bind<std::_Mem_fn<ReturnTypeT (ClassT::*)(Args ...)
 #elif defined _MSC_VER  // MS Visual Studio
 struct function_traits<
   std::_Binder<std::_Unforced, ReturnTypeT (ClassT::*)(Args ...) const, FArgs ...>>
+#elif defined _CPPLIB_VER  // Dinkumware C++ library
+struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT(__cdecl ClassT::*)(Args ...) const, FArgs ...>>
 #else
 #error "Unsupported C++ compiler / standard library"
 #endif
@@ -120,6 +124,8 @@ struct function_traits<std::__bind<ReturnTypeT( &)(Args ...), FArgs ...>>
 struct function_traits<std::_Bind<ReturnTypeT(*(FArgs ...))(Args ...)>>
 #elif defined _MSC_VER  // MS Visual Studio
 struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT( &)(Args ...), FArgs ...>>
+#elif defined _CPPLIB_VER  // Dinkumware C++ library
+struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT(__cdecl &)(Args ...), FArgs ...>>
 #else
 #error "Unsupported C++ compiler / standard library"
 #endif
-- 
2.25.1

