From 38e242f17c0cfd59d4506ae7992c2b93fe7eb79a Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Tue, 13 Jun 2023 11:20:08 +0200
Subject: [PATCH] added support for Dinkum C library

---
 rclcpp/include/rclcpp/function_traits.hpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/rclcpp/include/rclcpp/function_traits.hpp b/rclcpp/include/rclcpp/function_traits.hpp
index b6af0aeb..ff626a2b 100644
--- a/rclcpp/include/rclcpp/function_traits.hpp
+++ b/rclcpp/include/rclcpp/function_traits.hpp
@@ -91,6 +91,8 @@ struct function_traits<std::_Bind<std::_Mem_fn<ReturnTypeT (ClassT::*)(Args ...)
 #elif defined _MSC_VER  // MS Visual Studio
 struct function_traits<
   std::_Binder<std::_Unforced, ReturnTypeT (ClassT::*)(Args ...), FArgs ...>>
+#elif defined _CPPLIB_VER  // Dinkumware C++ library
+struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT(__cdecl ClassT::*)(Args ...), FArgs ...>>
 #else
 #error "Unsupported C++ compiler / standard library"
 #endif
@@ -110,6 +112,8 @@ struct function_traits<std::_Bind<std::_Mem_fn<ReturnTypeT (ClassT::*)(Args ...)
 #elif defined _MSC_VER  // MS Visual Studio
 struct function_traits<
   std::_Binder<std::_Unforced, ReturnTypeT (ClassT::*)(Args ...) const, FArgs ...>>
+#elif defined _CPPLIB_VER  // Dinkumware C++ library
+struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT(__cdecl ClassT::*)(Args ...) const, FArgs ...>>
 #else
 #error "Unsupported C++ compiler / standard library"
 #endif
@@ -126,6 +130,8 @@ struct function_traits<std::__bind<ReturnTypeT( &)(Args ...), FArgs ...>>
 struct function_traits<std::_Bind<ReturnTypeT(*(FArgs ...))(Args ...)>>
 #elif defined _MSC_VER  // MS Visual Studio
 struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT( &)(Args ...), FArgs ...>>
+#elif defined _CPPLIB_VER  // Dinkumware C++ library
+struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT(__cdecl &)(Args ...), FArgs ...>>
 #else
 #error "Unsupported C++ compiler / standard library"
 #endif
-- 
2.34.1

