diff --git a/rclcpp/include/rclcpp/executor.hpp b/rclcpp/include/rclcpp/executor.hpp
index 8471754..958764f 100644
--- a/rclcpp/include/rclcpp/executor.hpp
+++ b/rclcpp/include/rclcpp/executor.hpp
@@ -238,7 +238,11 @@ public:
     std::chrono::nanoseconds timeout_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
       timeout);
     if (timeout_ns > std::chrono::nanoseconds::zero()) {
+#if defined (__VXWORKS__)
+      end_time += std::chrono::duration_cast<std::chrono::milliseconds> (timeout_ns);
+#else
       end_time += timeout_ns;
+#endif
     }
     std::chrono::nanoseconds timeout_left = timeout_ns;
 
