diff --git a/rclcpp_components/CMakeLists.txt b/rclcpp_components/CMakeLists.txt
index 95ba47f..98bc8fa 100644
--- a/rclcpp_components/CMakeLists.txt
+++ b/rclcpp_components/CMakeLists.txt
@@ -16,6 +16,8 @@ find_package(class_loader REQUIRED)
 find_package(composition_interfaces REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rcpputils REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 include_directories(include)
 
@@ -39,6 +41,7 @@ add_executable(
 target_link_libraries(component_container component_manager)
 ament_target_dependencies(component_container
   "rclcpp"
+  "yaml"
 )
 
 add_executable(
@@ -48,6 +51,7 @@ add_executable(
 target_link_libraries(component_container_mt component_manager)
 ament_target_dependencies(component_container_mt
   "rclcpp"
+  "yaml"
 )
 
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
