diff --git a/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp b/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp
index d3b21e2..8643459 100644
--- a/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp
+++ b/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp
@@ -337,7 +337,8 @@ __declare_parameter_common(
   bool use_overrides = true)
 {
   using rclcpp::node_interfaces::ParameterInfo;
-  std::map<std::string, ParameterInfo> parameter_infos {{name, ParameterInfo()}};
+  std::map<std::string, ParameterInfo> parameter_infos = std::map<std::string, ParameterInfo>{};
+  parameter_infos.insert(std::make_pair(name, ParameterInfo()));
   parameter_infos.at(name).descriptor = parameter_descriptor;
 
   // Use the value from the overrides if available, otherwise use the default.
diff --git a/rclcpp_components/src/component_manager.cpp b/rclcpp_components/src/component_manager.cpp
index b7e8746..0c8953c 100644
--- a/rclcpp_components/src/component_manager.cpp
+++ b/rclcpp_components/src/component_manager.cpp
@@ -76,7 +76,8 @@ ComponentManager::get_component_resources(const std::string & package_name) cons
     if (!rcpputils::fs::path(library_path).is_absolute()) {
       library_path = base_path + "/" + library_path;
     }
-    resources.push_back({parts[0], library_path});
+    ComponentResource rsrc (parts[0], library_path);
+    resources.push_back(rsrc);
   }
   return resources;
 }
